src/libraries/System.Net.NameResolution/src/System/Net/NameResolutionMetrics.cs

```diff
@@ -13,15 +13,25 @@ internal static class NameResolutionMetrics
         private static readonly Meter s_meter = new("System.Net.NameResolution");
 
         private static readonly Histogram<double> s_lookupDuration = s_meter.CreateHistogram<double>(
-            name: "dns.lookups.duration",
+            name: "dns.lookup.duration",
             unit: "s",
             description: "Measures the time taken to perform a DNS lookup.");
 
         public static bool IsEnabled() => s_lookupDuration.Enabled;
 
-        public static void AfterResolution(TimeSpan duration, string hostName)
+        public static void AfterResolution(TimeSpan duration, string hostName, string? errorType)
```

Comment:
You're explicitly passing null for errorType in multiple places. You could make it optional here with a default null value and then avoid saying `, errorType: null` at each of those call sites.